browser/mapping.go

```diff
@@ -708,6 +743,54 @@ func mapBrowser(vu moduleVU, b api.Browser, wsURL string, isRemoteBrowser bool)
 	}
 }
 
+// getOrInitBrowser retrieves the browser for the iteration from the browser registry
+// if it is already initialized. Otherwise initializes a new browser for the iteration
+// and stores it in the registry.
+func getOrInitBrowser(
+	bt *chromium.BrowserType, vu moduleVU, wsURL string, isRemoteBrowser bool,
+) (api.Browser, error) {
+	// Index browser pool per VU-scenario-iteration
+	id := fmt.Sprintf("%d-%s-%d",
+		vu.State().VUID,
+		k6ext.GetScenarioName(vu.Context()),
+		vu.State().Iteration,
+	)
+
+	var (
+		ok  bool
+		err error
+		b   api.Browser
+	)
+
+	if b, ok = vu.getBrowser(id); ok {
```

Comment:
~~It looks like we're building a new `browser` object on every iteration, so this would always return `nil` and `false`, no? If that's the case then maybe we don't need this check.~~

Now i see why -- calls to the other browser APIs will retrieve the associated `browser` object (e.g. i can call `newContext` multiple times and only the first call to `newContext` will create the `browser` object).